src/components/manager/runs/RunLastContent.js

```diff
@@ -16,47 +17,111 @@ const COLUMNS_LAST_CONTENT = [
         title: "File Names",
         dataIndex: "fileNames",
         key: "fileNames",
-        render: (fileNames) => (
-            <>
-                {fileNames.map((fileName, index) => (
-                    <div key={index}>{fileName}</div>
-                ))}
-            </>
-        ),
+        render: (fileNames, record) => <FileNamesCell fileNames={fileNames} dataType={record.dataType}/>,
     },
 ];
 
+const modalListStyle = {
+    whiteSpace: "nowrap",
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+};
+function FileNamesCell({fileNames, dataType}) {
+    const [isModalVisible, setIsModalVisible] = useState(false);
+
+    const isTruncated = fileNames.length > 4;
+    const truncatedFileNames = isTruncated
+        ? [...fileNames.slice(0, 2), <Icon type="more" key="more-icon"/>, ...fileNames.slice(-2)]
+        : fileNames;
+
+    const divStyle = isTruncated
+        ? {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            justifyContent: "center",
+            cursor: "pointer",
+        }
+        : {};
+
+    const openModal = () => isTruncated && setIsModalVisible(true);
+
+    const closeModal = () => setIsModalVisible(false);
+
+    return (
+        <>
+            <div onClick={openModal} style={divStyle}>
+                {truncatedFileNames.map((element, index) =>
+                    typeof element === "string" ? <div key={index}>{element}</div> : element,
+                )}
+            </div>
+            <Modal
+                title={`${dataType.charAt(0).toUpperCase() + dataType.slice(1)} files`}
+                footer={null}
+                visible={isModalVisible}
+                onCancel={closeModal}
+                bodyStyle={{maxHeight: "80vh", overflowY: "auto", whiteSpace: "text-overflow"}}
+            >
+                 <ul style={{ padding: "0 20px", listStyle: "none" }}>
+                    {fileNames.map((fileName, index) => (
+                        <li key={index} style={modalListStyle}>- {fileName}</li>
+                    ))}
+                </ul>
+            </Modal>
+        </>
+    );
+}
+
+FileNamesCell.propTypes = {
+    fileNames: PropTypes.arrayOf(PropTypes.string).isRequired,
+    dataType: PropTypes.string.isRequired,
+};
+
 const buildKeyFromRecord = (record) => `${record.dataType}-${record.tableId}`;
 
+const fileNameFromPath = (path) => path.split("/").at(-1);
+
+const getFileInputsFromWorkflowMetadata = (workflowMetadata) => {
+    return workflowMetadata.inputs
+        .map(input => `${workflowMetadata.id}.${input.id}`);
```

Comment:
not all inputs are of type file - check the input type and filter to those which start with file.